src/Parser/TemplateParser.cs

```diff
@@ -138,22 +139,90 @@ private string ParseName(out int parameterIndex)
                 c = Peek();
                 if (c == '}' || c == ':' || c == ',')
                     parameterIndex = parsed;
-                _position = start;        
+                _position = start;
             }
 
             if (parameterIndex == -1)
                 _isPositional = false;
 
-            return ReadUntil(HoleDelimiters);            
+            return ReadUntil(HoleDelimiters);
         }
 
+        /// <summary>
+        /// Parse format after hole name/index. Handle the escaped { and } in the format. Don't read the last }
+        /// </summary>
+        /// <returns></returns>
         private string ParseFormat()
         {
+
             Skip(':');
-            // TODO: Escaped }} in formats?
-            return ReadUntil('}');
+            string format = null;
+            while (true)
+            {
+                if (format == null)
+                {
+                    //avoid string composition.
+                    format = ReadUntil(TextDelimiters);
+                }
+                else
+                {
+                    format += ReadUntil(TextDelimiters);
+                }
+
+                var c = Read();
+
+                switch (c)
+                {
+                    case '}':
+                        {
+                            if (_position == _length)
+                            {
+                                //unread the }
+                                _position--;
+                                //end of template so done
+                                return format;
+                            }
+
+                            var next = Peek();
+                            if (next == '}')
+                            {
+                                //this is an escaped } and need to be added to the format.
+                                Skip('}');
+                                format += "}";
+                            }
+                            else
+                            {
+                                //unread the }
+                                _position--;
+                                //done
+                                return format;
+                            }
+                            break;
+                        }
+                    case '{':
+                        {
+                            //we need a second {, otherwise this format is wrong.
+                            var next = Peek();
```

Comment:
We don't really need the local here. You can do `if (Peek() == '{')` and then remove the braces around the `case` code, unless that's the style you always use?